userdata.sh.tmpl

```diff
@@ -14,8 +14,8 @@ sudo yum-config-manager --add-repo https://pkgs.tailscale.com/stable/amazon-linu
 sudo yum install -y tailscale
 
 %{ if tailscaled_extra_flags_enabled == true }
-echo "Exporting FLAGS to environment variable..."
-export FLAGS=${tailscaled_extra_flags}%
+echo "Exporting FLAGS to /etc/default/tailscaled..."
+sudo sed -i "s|^FLAGS=.*|FLAGS=\"${tailscaled_extra_flags}\"|" /etc/default/tailscaled
```

Comment:
Just curious, why is this needed?

Reply:
To control where state (including preferences and keys) is stored, we use flag `--state=`.
On Linux, setting tailscaled flags is configured by modifying the FLAGS in `/etc/default/tailscaled`, which the systemd unit definition will include.

Originally this file looks like this:
```sh
# Set the port to listen on for incoming VPN packets.
# Remote nodes will automatically be informed about the new port number,
# but you might want to configure this in order to set external firewall
# settings.
PORT="41641"

# Extra flags you might want to pass to tailscaled.
FLAGS=""
```

Reply:
👍